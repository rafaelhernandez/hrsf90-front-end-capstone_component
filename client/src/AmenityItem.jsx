import React from 'react';

const AmenityItem = (props) => (
  <div className="div-description-amenity-item">
    {props.amenity}
  </div>
);

export default AmenityItem;
