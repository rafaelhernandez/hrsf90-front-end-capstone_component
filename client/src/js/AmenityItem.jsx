import React from 'react';
import SVGIcons from './SVGIcons.js';

const AmenityItem = (props) => (
  <div className="div-description-amenity-item">
    <div className="div-description-characteristics-block">
      <div className="div-description-characteristics-item">
        <div className="div-description-characteristics-table-cell">
          <span className="div-description-characteristics-table-cell-margin" dangerouslySetInnerHTML={{ __html: SVGIcons[props.amenity] }} ></span>
        </div>
        <div className="div-description-characteristics-amenity-cell">
          {props.amenity}
        </div>
      </div>
    </div>
  </div>
);

export default AmenityItem;
