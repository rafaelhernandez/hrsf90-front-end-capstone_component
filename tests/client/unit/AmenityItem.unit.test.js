import React from 'react';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import AmenityItem from '../../../client/src/AmenityItem.jsx';

configure({ adapter: new Adapter() });

describe('AmenityItem', () => {
  it('should render correctly', () => {
    const output = shallow(
      <AmenityItem amenity="mockAmenity1"></AmenityItem>
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });
});
