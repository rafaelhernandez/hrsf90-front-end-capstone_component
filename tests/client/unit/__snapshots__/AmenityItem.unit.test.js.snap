// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AmenityItem should render correctly 1`] = `
<div
  className="div-description-amenity-item"
>
  mockAmenity1
</div>
`;
